### PR TITLE
[ESI-11793] Distinguish DB fetch resource not found error

### PR DESCRIPTION
**Pull request by Fenthick**
* DB fetch NotFound case is distinguished from other errors (JS extensions can act based on that information)
* pass all error details in DB fetch error string

### CODE REVIEW

**Comment by Fenthick** [comment]
@marcin-ptaszynski @p-kozlowski @michalzurawski @przemyslaw-dobrowolski-cl

**Comment by marcin-ptaszynski** [comment]
@Fenthick  I think we should also integrate it to [js layer of errors](https://github.com/cloudwan/gohan/blob/master/extension/otto/gohan_core.go#L113), so it actually causes 404 response. From what I currently see, it doesn't behave like this, or am I missing something?

File: db/errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package errors
+
+import stderrors "errors"
+
+var (
+	// NoSchemasInManager is returned when there is no schemas in schema manager
+	NoSchemasInManager = stderrors.New("No schemas in Manager. Did you remember to load them?")
+
+	// ResourceNotFound is returned when requested resource was not found
+	ResourceNotFound = stderrors.New("resource not found")
```
Comment (p-kozlowski) [inline db/errors/errors.go]: InternalError is returned when internal error happened.

Sorry, I could not resist. It's 2017, we've got modern IDEs, why do we still write comments?

Comment (Fenthick) [inline db/errors/errors.go]: Obvious comments removed.
Golint shows it as an error. Unfortunately it doesn't have an option to remove that check.

**Comment by Fenthick** [comment]
@marcin-ptaszynski 
There is also PR in extension code: https://bitbucket.org/ntti3/esi-wan-gohan/pull-requests/670/wip-esi-11793-handle-db_fetch-notfound/diff

**Comment by Fenthick** [comment]
Branch rebased.

**Comment by Fenthick** [comment]
@nati Could you take a look at this PR? It has been already approved by other reviewers.

**Pull request merged**